lib/json_schemer/configuration.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module JSONSchemer
+  class Configuration
+    module Defaults
+      BASE_URI = URI('json-schemer://schema').freeze
+      FORMATS = {}.freeze
+      CONTENT_ENCODINGS = {}.freeze
+      CONTENT_MEDIA_TYPES = {}.freeze
+      KEYWORDS = {}.freeze
+      BEFORE_PROPERTY_VALIDATION = [].freeze
+      AFTER_PROPERTY_VALIDATION = [].freeze
+      INSERT_PROPERTY_DEFAULTS = false
+      PROPERTY_RESOLVER = proc do |instance, property, results_with_tree_validity|
+        unless results_with_tree_validity.size == 1
+          results_with_tree_validity = results_with_tree_validity.select(&:last)
+        end
+        annotations = results_with_tree_validity.to_set { |result, _tree_valid| result.annotation }
+
+        if annotations.size == 1
+          instance[property] = annotations.first.clone
+          true
+        else
+          false
+        end
+      end
+      REF_RESOLVER = proc { |uri| raise UnknownRef, uri.to_s }
+      REGEXP_RESOLVER = 'ruby'
+      OUTPUT_FORMAT = 'classic'
+      RESOLVE_ENUMERATORS = false
+      ACCESS_MODE = nil
+    end
+
+    attr_accessor(
+      :base_uri,
+      :formats,
+      :content_encodings,
+      :content_media_types,
+      :keywords,
+      :before_property_validation,
+      :after_property_validation,
+      :insert_property_defaults,
+      :property_default_resolver,
+      :original_ref_resolver,
+      :original_regexp_resolver,
+      :output_format,
+      :resolve_enumerators,
+      :access_mode
+    )
+
+    def initialize
```

Comment:
@davishmcclurg So at this point I moved a bunch of options to the `Configuration` singleton, let me know if you have any feedback here. Did I miss any options that should be configurable here or vice versa?

Reply:
I commented below, but I think we probably want `meta_schema` and `format` as well (and `vocabulary` for completeness?)